packages/model-viewer/src/three-components/gltf-instance/correlated-scene-graph.ts

```diff
@@ -122,35 +149,46 @@ export class CorrelatedSceneGraph {
     const defaultMaterial = {name: 'Default'} as Material;
     const defaultReference = {type: 'materials', index: -1} as GLTFReference;
 
+    // Gets the parser cache.
+    const parserCache =
+        new ParserCache(cloneThreeGLTF.parser as unknown as ParserInterface);
+
     for (let i = 0; i < originalThreeGLTF.scenes.length; i++) {
       this[$parallelTraverseThreeScene](
           originalThreeGLTF.scenes[i],
           cloneThreeGLTF.scenes[i],
           (object: ThreeSceneObject, cloneObject: ThreeSceneObject) => {
-            let elementReference =
+            const elementReference =
                 upstreamCorrelatedSceneGraph.threeObjectMap.get(object);
 
-            if (elementReference == null) {
+            const getMaterial = (mesh: Mesh) => {
+              return mesh.material as MeshStandardMaterial;
+            };
+            // Only sets up default material if the mesh references it.
+            if ((object as Mesh).isMesh &&
+                getMaterial(object as Mesh).uuid ===
+                    parserCache.defaultMaterialUUID) {
               if (defaultReference.index < 0) {
                 if (cloneGLTF.materials == null) {
                   cloneGLTF.materials = [];
                 }
                 defaultReference.index = cloneGLTF.materials.length;
                 cloneGLTF.materials.push(defaultMaterial);
               }
-              elementReference = defaultReference;
             }
 
-            const {type, index} = elementReference;
-            const cloneElement = cloneGLTF[type]![index];
+            if (elementReference != null) {
```

Comment:
How about an early return instead to keep the indenting down?

Reply:
Done.